cmd/awsDataGather/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/red-hat-storage/ocs-osd-deployer/pkg/aws"
+	"github.com/red-hat-storage/ocs-osd-deployer/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	namespace, found := os.LookupEnv("NAMESPACE")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"NAMESPACE environment variable not found\n")
+		os.Exit(1)
+	}
+	deploymentName, found := os.LookupEnv("DEPLOYMENT_NAME")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"NAME environment variable not found\n")
+		os.Exit(1)
+	}
+
+	log.Info("Setting up k8s client")
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(options.Scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	// This will later be used as the OwnerReference for the data ConfigMap
+	var deployment appsv1.Deployment
+	utils.Retry(5, 1*time.Second, func() error {
+		err := k8sClient.Get(context.Background(), client.ObjectKey{
+			Name:      deploymentName,
+			Namespace: namespace,
+		}, &deployment)
+		return err
+	})
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Failed to find deployment '%s' in namespace '%s'", deploymentName, namespace))
+		os.Exit(1)
+	}
+
+	log.Info("Gathering AWS data")
+	if err := gatherData(aws.IMDSv1Server, deployment, k8sClient); err != nil {
+		log.Error(err, "Failed to gather AWS data")
+		os.Exit(1)
+	}
+
+	log.Info("AWS data gathering successfully completed!")
+
+	// Set up signal handler, so that a clean up procedure will be run if the pod running this code is deleted.
+	// This code is run as part of a deployment that is pushed to Kubernetes through an OLM Bundle.
+	// At this time, OLM Bundles do not support restartPolicy: OnFailure,
+	// so terminating this program will cause the pod to keep getting restarted and needlessly running.
+	// To avoid this, a signal handler is set up to only terminate the program when the pod is getting deleted.
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGTERM)
+	<-signalChan
+
+	log.Info("AWS data gather program terminating")
+}
+
+func gatherData(imdsServer string, deployment appsv1.Deployment, k8sClient client.Client) error {
```

Comment:
- maybe func rename is suited better as it's also creating configmap?
```suggestion
func gatherAndSaveData(imdsServer string, deployment appsv1.Deployment, k8sClient client.Client) error {
```